packages/mask/src/extension/popups/pages/Wallet/index.tsx

```diff
@@ -109,7 +110,7 @@ export default function Wallet() {
                 providerType: ProviderType.MaskWallet,
             })
         } else if (wallet?.owner) {
-            setChainId(ChainId.Mumbai)
+            SmartPayBundler.getSupportedChainId().then((chainId) => setChainId(chainId))
```

Comment:
Having a context provider and making async requests in the context is better. Avoid handling too much biz logic in UI components.